enterprise/server/backends/gcs_cache/gcs_cache.go

```diff
@@ -60,18 +61,28 @@ func NewGCSCache(bucketName, projectID string, ageInDays int64, opts ...option.C
 	return g, nil
 }
 
+func (g *GCSCache) bucketExists(ctx context.Context, bucketName string) (bool, error) {
+	_, spn := tracing.StartSpan(ctx)
+	defer spn.End()
+	_, err := g.gcsClient.Bucket(bucketName).Attrs(ctx)
+	return err == nil, err
+}
 func (g *GCSCache) createBucketIfNotExists(ctx context.Context, bucketName string) error {
-	if _, err := g.gcsClient.Bucket(bucketName).Attrs(ctx); err != nil {
+	if exists, _ := g.bucketExists(ctx, bucketName); !exists {
 		log.Printf("Creating storage bucket: %s", bucketName)
 		g.bucketHandle = g.gcsClient.Bucket(bucketName)
+		_, spn := tracing.StartSpan(ctx)
+		defer spn.End()
 		return g.bucketHandle.Create(ctx, g.projectID, nil)
 	}
 	g.bucketHandle = g.gcsClient.Bucket(bucketName)
 	return nil
 }
 
 func (g *GCSCache) setBucketTTL(ctx context.Context, bucketName string, ageInDays int64) error {
+	_, spn := tracing.StartSpan(ctx)
 	attrs, err := g.gcsClient.Bucket(bucketName).Attrs(ctx)
+	spn.End()
```

Comment:
can we override ctx with the ctx returned from StartSpan so spans get correctly nested?

	ctx, spn := tracing.StartSpan(ctx)
	attrs, err := g.gcsClient.Bucket(bucketName).Attrs(ctx)
	spn.End()

Reply:
Yep, done.